pkg/agent/client.go

```diff
@@ -237,12 +238,17 @@ func (a *Client) Connect() (int, error) {
 		conn.Close() /* #nosec G104 */
 		return 0, err
 	}
-	serverID, err := serverID(stream)
+	md, err := getAndValidateStreamHeader(stream)
```

Comment:
`conn.Close()` and returning 0 and error called in multiple times in this function. I tried to use the named return type and defer function to cover these cases but considering that makes the code less easy to be maintainable, I copied and pasted the error handling here.